worker-document/src/main/java/com/hpe/caf/worker/document/scripting/JavaScriptEngine.java

```diff
@@ -24,24 +26,41 @@
 import javax.script.ScriptEngine;
 import javax.script.ScriptEngineManager;
 import javax.script.ScriptException;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.HostAccess;
 
 public final class JavaScriptEngine implements ObjectCodeProvider
 {
-    private final ScriptEngine scriptEngine;
-    private final Bindings scriptEngineBindings;
+    private final ScriptEngine graalEngine;
+    private final Bindings graalEngineBindings;
+    private final ScriptEngine nashornEngine;
+    private final Bindings nashornEngineBindings;
     private final Object scriptEngineBindingsLock;
 
     public JavaScriptEngine()
```

Comment:
This is always going to load both JavaScript engines.  It would be better to have separate `NashornJSEngine` and a `GraalJSEngine` classes, and only construct them if they are used.  Currently the `JavaScriptEngineLazy` class ensures that the `JavaScriptEngine` is only constructed if it is used (because it is expensive) but the way you've changed this if either engine is used both are loaded.